### PR TITLE
Update index.js

### PR DESCRIPTION
**Pull request by xerosanyam**
1. When using gmail, lesssecureapps & DisplayUnlockCaptcha needs to be enabled.
2. Sometimes displayName is not received, sending Hey *undefined*! in email.

### CODE REVIEW

File: quickstarts/email-users/functions/index.js [hunk 1]
```diff
@@ -66,13 +69,16 @@ exports.sendByeEmail = functions.auth.user().onDelete(event => {
...
     to: email
   };
 
   // The user subscribed to the newsletter.
   mailOptions.subject = `Welcome to ${APP_NAME}!`;
-  mailOptions.text = `Hey ${displayName}!, Welcome to ${APP_NAME}. I hope you will enjoy our service.`;
+  if(!displayName){
+	displayName=''
+  }
+  mailOptions.text = `Hey ${displayName}! Welcome to ${APP_NAME}. I hope you will enjoy our service.`;
```
Comment (nicolasgarnier) [inline quickstarts/email-users/functions/index.js]: You can replace the 4 lines above by:

```javascript
mailOptions.text = `Hey ${displayName || ''}! Welcome to ${APP_NAME}. I hope you will enjoy our service.`;
```

File: quickstarts/email-users/functions/index.js [hunk 2]
```diff
@@ -40,7 +43,7 @@ exports.sendWelcomeEmail = functions.auth.user().onCreate(event => {
   const user = event.data; // The Firebase user.
 
   const email = user.email; // The email of the user.
-  const displayName = user.displayName; // The display name of the user.
+  var displayName = user.displayName; // The display name of the user.
```
Comment (nicolasgarnier) [inline quickstarts/email-users/functions/index.js]: switch back to `const` (see my comment below)

File: quickstarts/email-users/functions/index.js [hunk 3]
```diff
@@ -57,7 +60,7 @@ exports.sendByeEmail = functions.auth.user().onDelete(event => {
   const user = event.data;
 
   const email = user.email;
-  const displayName = user.displayName;
+  var displayName = user.displayName;
```
Comment (nicolasgarnier) [inline quickstarts/email-users/functions/index.js]: switch back to `const` (see my comment below)

File: quickstarts/email-users/functions/index.js [hunk 4]
```diff
@@ -81,13 +87,16 @@ function sendWelcomeEmail(email, displayName) {
...
     to: email
   };
 
   // The user unsubscribed to the newsletter.
   mailOptions.subject = `Bye!`;
-  mailOptions.text = `Hey ${displayName}!, We confirm that we have deleted your ${APP_NAME} account.`;
+  if(!displayName){
+	displayName=''
+  }
+  mailOptions.text = `Hey ${displayName}! We confirm that we have deleted your ${APP_NAME} account.`;
```
Comment (nicolasgarnier) [inline quickstarts/email-users/functions/index.js]: Same as above.

File: quickstarts/email-users/functions/index.js (hunk 1 above)
Comment (xerosanyam) [inline quickstarts/email-users/functions/index.js]: Neat. :+1:

**Comment by nicolasgarnier** [comment]
Just applied the changes to your branch.

**Comment by nicolasgarnier** [comment]
Thanks for the fixes @xerosanyam !

**Comment by xerosanyam** [comment]
Sure @nicolasgarnier . Happy to help :)

**Pull request merged**